store/store.go

```diff
@@ -735,6 +736,16 @@ type LinkMetadataStore interface {
 	Get(url string, timestamp int64) (*model.LinkMetadata, error)
 }
 
+// AtomicStore is an interface for key/value store with atomic operations
+type AtomicStore interface {
+	SaveOrUpdate(keyVal *model.AtomicKeyValue) (*model.AtomicKeyValue, *model.AppError)
+	CompareAndSet(keyVal *model.AtomicKeyValue, oldValue []byte) (bool, *model.AppError)
+	CompareAndDelete(key string, oldValue []byte) (bool, *model.AppError)
+	Get(key string) (*model.AtomicKeyValue, *model.AppError)
+	Delete(key string) *model.AppError
+	DeleteAllExpired() *model.AppError
```

Comment:
We use normal errors in the store now. Please take a look at the other stores to see examples.